### ISSUE TITLE
"Play previous" command not working for Spotify is not working as expected #01

### ISSUE THREAD

**Issue by vandnakapoor19** [mozilla/firefox-voice#1497]
**Prerequisites:**
Mic permissions and autoplay are enabled.
Have a Spotify or Youtube tab open and active. Have several songs/videos played in the tab?

**STR:**
Click on the mic icon from the browser toolbar.
Say or write a "Play previous" command.
Expected result:
The previous song/video starts playing.

**Actual result:**
It skips previous songs and plays song before of previous.

**Notes:**

System Configuration: -
OS: Windows 10x64
Firefox Nightly: Version 75.0a1
Node: v12.16.1

![Play_previous_err](https://user-images.githubusercontent.com/45131874/78713114-ba80c080-78e7-11ea-88d5-fc805b998734.gif)

**Comment by vandnakapoor19** [comment]
I've fixed the issue and made my pull request 

https://github.com/mozilla/firefox-voice/pull/1499

**Comment by alexandra-martin** [comment]
From what I remember "Play previous" has in the code the option to go to previous 2 times, because normally when clicking the previous button on Spotify, Soundcloud and Youtube (when in a playlist) it starts the played song/video again, if it has played more than some seconds. 
Currently if the music is at the beginning it goes back 2 songs, but if the music has played more along the way, the command goes correctly to the previous song. This applies only to Spotify and Soundcloud. Youtube uses the "Back" browser button when doing this command.

![7](https://user-images.githubusercontent.com/54401091/78797860-f3d92f00-79c0-11ea-99bd-8049cb3eeba3.gif)

**Comment by melvin2016** [comment]
@vandnakapoor19 @alexandra-martin 
After playing with the Spotify player I understood that the previous button works in 2 ways:

1. if the previous button is clicked before reaching the 0:29 seconds the song goes back to the previous song with one click.

2. if the song goes beyond 0:29 seconds we need to click twice.

These are my assumptions I don't know if this is the correct way.  🙂

**Comment by vandnakapoor19** [comment]
@melvin2016 Yes, you are correct! I've to just make a change in the logic where we are just checking for 0.03 seconds.
